### DISCUSSION TITLE
Fields BT-538-Lot BT-536-Lot BT-537-Lot BT-36-Lot mandatory or not ?

### DISCUSSION THREAD

**Discussion by jcagnolo** [OP-TED/eForms-SDK#831] [Q&A] [answered]
Hello everyone,
 
I would like to report a problem in the fields listed below:
 
- Other duration (BT-538-Lot) => mandatory (Not mandatory for schematrons)
- Start date (BT-536-Lot) => mandatory under specific conditions
- End date (BT-537-Lot) => mandatory (Not required for schematrons)
- Duration (BT-36-Lot) => mandatory (Not mandatory for schematrons)
 
Schematrons do not report any errors on these 4 fields when they are not filled.
 
Moreover, BT-536-Lot is conditionally mandatory, whereas BT-537-Lot is unconditionally mandatory. As a result, the latter has an asterisk to inform the user that the field is mandatory, unlike BT-536-Lot.
 
How did you manage this issue?

Thank's

**Comment by YvesJo** [comment, marked as answer]
Hi,
I will try to provide a first answer to your question, knowing the SDK version and the Notice Subtype you are referring to would however help focus.

Following the Regulation (extended) Annex, for many forms (i.e. Notice Subtypes), this information is purely optional and for some other forms, it is Conditionally Mandatory.
![image](https://github.com/OP-TED/eForms-SDK/assets/42802091/f873728f-6267-49da-986b-d8d9b8cebe74)

Conditional rules have been activated with SDK 1.8.

With SDK 1.10 a simplified, more flexible approach has been adopted and the possible combinations are [documented](https://docs.ted.europa.eu/eforms/latest/schema/procedure-lot-part-information.html#plannedPeriodSection).

With "BT-537-Lot is unconditionally mandatory", I understand it would always be mandatory. It is neither the case in SDK 1.10 nor in SDK 1.9
In fields.json of SDK 1.9 you may find the following:

    _"id" : "BT-537-Lot",
    "parentNodeId" : "ND-LotDuration",
    "name" : "Duration End Date",
     ...
    "forbidden" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "1", "2", "3", "4", "5", "6", "23", "24", "36", "37", "CEI", "X01", "X02" ],
        "value" : true,
        "severity" : "ERROR"
      }, {
        "noticeTypes" : [ "7", "8", "9", "10", "11" , "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "25", "26", "27", "28", "29", "30", "31", "32", "33", "34", "35", "38", "39", "40", "T01", "T02" ],
        "condition" : "{ND-LotDuration} ${(BT-36-Lot is present) or (BT-538-Lot is present)}",
        "value" : true,
        "severity" : "ERROR"
      } ]
    },
    "mandatory" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "7", "8", "9", "10", "11", "15", "16", "17", "18", "19", "22", "T01", "T02" ],
        "value" : true,
        "severity" : "ERROR"
      } ]
    },_


  "
All the Notice Subtypes listed under "mandatory" are also listed under "forbidden" and they share the same condition (the condition contraposition has to be used for the mandatory rule)

The condition for the forbidden part state that if BT-36 or BT-538 is present, then BT-537 is forbidden. For the mandatory part, the condition for BT-537 to be mandatory for the listed Notice Subtypes is "neither BT-36, nor BT-538 is present", so for these forms BT-537 is mandatory in absence of BT-36 and BT-538.

The rules shouldn't be evaluated at the very start as information is missing as long as no information is provided. May be this is the situation faced in the described case?

KR

**Comment by jcagnolo** [comment]
Thank you for your reply. 
I'll pass it on to our developers and get back to you if I need any clarification.
